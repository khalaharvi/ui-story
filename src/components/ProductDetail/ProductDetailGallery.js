import React from 'react';
import ImageGallery from 'react-image-gallery';
import PropTypes from 'prop-types';

import { cx } from './ProductDetail';

const ProductDetailGallery = ({ images }) => (
  <div className={cx('gallery')}>
    <ImageGallery
      items={images}
      showNav={false}
      showPlayButton={false}
      showFullscreenButton={false}
      showThumbnails={images.length > 1}
    />
  </div>
);

ProductDetailGallery.propTypes = {
  images: PropTypes.arrayOf(PropTypes.shape({
    original: PropTypes.string.isRequired,
    thumbnail: PropTypes.string.isRequired
  }))
};

export default ProductDetailGallery;
