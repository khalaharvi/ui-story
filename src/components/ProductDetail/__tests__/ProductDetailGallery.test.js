import React from 'react';
import { shallow } from 'enzyme';

import ProductDetailGallery from '../ProductDetailGallery';

describe('<ProductDetailGallery />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProductDetailGallery
        images={[
          {
            original: 'https://placehold.it/600x600',
            thumbnail: 'https://placehold.it/100x100'
          },
          {
            original: 'https://placehold.it/600x600',
            thumbnail: 'https://placehold.it/100x100'
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
