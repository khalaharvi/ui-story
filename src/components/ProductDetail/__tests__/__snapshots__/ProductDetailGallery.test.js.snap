// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailGallery /> renders correctly 1`] = `
<div
  className="dh-product-detail-gallery"
>
  <ImageGallery
    autoPlay={false}
    disableArrowKeys={false}
    disableSwipe={false}
    disableThumbnailScroll={false}
    flickThreshold={0.4}
    indexSeparator=" / "
    infinite={true}
    items={
      Array [
        Object {
          "original": "https://placehold.it/600x600",
          "thumbnail": "https://placehold.it/100x100",
        },
        Object {
          "original": "https://placehold.it/600x600",
          "thumbnail": "https://placehold.it/100x100",
        },
      ]
    }
    lazyLoad={false}
    preventDefaultTouchmoveEvent={false}
    renderFullscreenButton={[Function]}
    renderLeftNav={[Function]}
    renderPlayPauseButton={[Function]}
    renderRightNav={[Function]}
    showBullets={false}
    showFullscreenButton={false}
    showIndex={false}
    showNav={false}
    showPlayButton={false}
    showThumbnails={true}
    slideDuration={450}
    slideInterval={3000}
    startIndex={0}
    stopPropagation={false}
    swipeThreshold={30}
    swipingTransitionDuration={0}
    thumbnailPosition="bottom"
    useBrowserFullscreen={true}
    useTranslate3D={true}
  />
</div>
`;
